modApiServer/src/org/aion/api/server/ApiAion.java

```diff
@@ -82,7 +80,7 @@
 
     // delegate concurrency to underlying object
     protected static NrgOracle NRG_ORACLE;
-    protected IChainInstancePOW ac; // assumption: blockchainImpl et al. provide concurrency guarantee
+    protected IGenericAionChain ac; // assumption: blockchainImpl et al. provide concurrency guarantee
```

Comment:
Please rename this protected field and the constructor parameter to something more suitable and also the other occurrences of the same problem in the other classes.